test/RestrictedNORI.test.ts

```diff
@@ -0,0 +1,681 @@
+      const retrievedRestrictionDuration =
+        await rNori.getRestrictionDurationForMethodologyAndVersion(1, 0);
+      expect(retrievedRestrictionDuration.wasSet).to.equal(true);
+      expect(retrievedRestrictionDuration.duration).to.equal(
+        SECONDS_IN_10_YEARS
+      );
+    });
+    it('should be able to set and get a new restriction duration for a given methodology and version', async () => {
+      const { rNori } = await setupTestLocal({});
+      const methodology = 2;
+      const methodologyVersion = 0;
+      const newDuration = SECONDS_IN_1_YEAR_AVG;
+      const scheduleDurationBeforeSetting =
+        await rNori.getRestrictionDurationForMethodologyAndVersion(
+          methodology,
+          methodologyVersion
+        );
+      expect(scheduleDurationBeforeSetting.wasSet).to.equal(false);
+      expect(scheduleDurationBeforeSetting.duration).to.equal(0);
+      await rNori.setRestrictionDurationForMethodologyAndVersion(
+        methodology,
+        methodologyVersion,
+        newDuration
+      );
+      const scheduleDurationAfterSetting =
+        await rNori.getRestrictionDurationForMethodologyAndVersion(
+          methodology,
+          methodologyVersion
+        );
+      expect(scheduleDurationAfterSetting.wasSet).to.equal(true);
+      expect(scheduleDurationAfterSetting.duration).to.equal(
+        SECONDS_IN_1_YEAR_AVG
+      );
+    });
+    it('should revert if the transaction sender does not have the DEFAULT_ADMIN_ROLE', async () => {
+      const { rNori } = await setupTestLocal({});
+      const methodology = 2;
+      const methodologyVersion = 0;
+      const newDuration = SECONDS_IN_1_YEAR_AVG;
+      await expect(
+        rNori
+          .connect(hre.namedSigners.buyer) // funded address without DEFAULT_ADMIN_ROLE
+          .setRestrictionDurationForMethodologyAndVersion(
+            methodology,
+            methodologyVersion,
+            newDuration
+          )
+      ).to.be.reverted;
+    });
+    it('should revert if a restriction schedule is being created for a methodology/version that does not have a duration set', async () => {
+      const { removal, fifoMarket, hre } = await setupTestLocal({});
+      const removalIdWithMethodology2 = await createRemovalTokenId({
+        removal,
+        removalData: { methodology: 2 },
+        hre,
+      });
+      const projectId = 1_234_567_890;
+      const scheduleStartTime = await getLatestBlockTime({ hre });
+      const amount = 20_000_000;
+      const packedData = await createBatchMintData({
+        hre,
+        fifoMarket,
+        listNow: true,
+        projectId,
+        scheduleStartTime,
+      });
+
+      await expect(
+        removal.mintBatch(
+          hre.namedAccounts.supplier,
+          [amount],
+          [removalIdWithMethodology2],
+          packedData
+        )
+      ).to.be.reverted;
+      // ).to.be.revertedWith(`RestrictionDurationNotSet(${projectId})`);
```

Comment:
ah yeah.... so this is something I mentioned up above about how when I moved schedule creation upon list into the Market contract's `tokensReceived`, it seems that any revert that occurs during the execution of that function actually causes the revert message `'ERC1155: transfer to non ERC1155Receiver implementer'` . You can see that's true with the updated test.  There's something weird about control flow with that hook... I actually had to add a whole new test that creates a schedule directly with a call to `createSchedule` in order to get a gas report for that specific function, because when `createSchedule` is called in the course of executing `tokensReceived`, it doesn't even show up in the gas reporter.  Other errors and failures that I've seen in the course of that hook also come up with this same revert string. Not sure why things are working this way but there seems to be something exceptional about the hook.

Reply:
per our discussion and investigation: using custom errors in a try/catch context that's expecting an `Error` such as in ERC1155Upgradeable.sol 
```
            try IERC1155ReceiverUpgradeable(to).onERC1155Received(operator, from, id, amount, data) returns (bytes4 response) {
                if (response != IERC1155ReceiverUpgradeable.onERC1155Received.selector) {
                    revert("ERC1155: ERC1155Receiver rejected tokens");
                }
            } catch Error(string memory reason) {
                revert(reason);
            } catch {
                revert("ERC1155: transfer to non ERC1155Receiver implementer");
            }
```
Custom errors aren't caught.  So I changed things in the `createSchedule` function to use revert strings instead of custom errors and left a note in the natspec about why that's happening.  We may find there are other places in hooks like this where we want to update how errors are handled.